openproblems/data/tabula_muris_senis.py

```diff
@@ -73,20 +58,31 @@ def load_tabula_muris_senis(test=False, method_list=None, organ_list=None):
     and droplet-fat anndata sets. (no facs-fat dataset available)
     """
 
-    # df containing figshare links, method of collection, and organ for each
-    # tabula muris dataset
-    url_df = pd.read_csv(
-        os.path.join(
-            os.path.dirname(__file__),
-            "tabula_muris_senis_data_objects",
-            "tabula_muris_senis_data_objects.csv",
-        ),
-        header=0,
-    )
+    if not method_list:
+        raise ValueError("`method_list' argument not provided")
+    unknown_methods = set(method_list) - set(["facs", "droplet"])
+    if unknown_methods:
+        raise ValueError(
+            f"Unknown methods provided in `method_list': {','.join(unknown_methods)}. "
+            "Known methods are `facs' and `droplet'"
+        )
+    organ_list = [x.lower() for x in organ_list]
+
+    datasets_path = f"/curation/v1/collections/{COLLECTION_ID}"
+    url = f"{API_BASE}{datasets_path}"
+    res = requests.get(url=url)
+    datasets = res.json()["datasets"]
+
+    adata_list = []
+    for dataset in datasets:
+        if matching_dataset(dataset, method_list, organ_list):
+            adata_list.append(load_raw_counts(dataset))
+
+    if len(adata_list) > 1:
+        adata = adata_list[0].concatenate(adata_list[1:], join="outer")
+    else:
+        adata = adata_list[0]
 
-    subset_df = get_filenames_and_urls(url_df, method_list, organ_list)
-    adata_list = make_anndata_list(subset_df, test)
-    adata = adata_list[0].concatenate(adata_list[1:])
     if test:
         sc.pp.subsample(adata, n_obs=500)
         adata = adata[:, :1000]
```

Comment:
Do we need to be more specific about test subsampling here? This might be used in a multi-batch setting... Could subsample per covariate if needed.

Reply:
See `openproblems.data.utils.subsample_even`. Currently only takes one covariate but we could make an amalgamation of covariates by simply concatenating them.

Reply:
I think this is only for running tests, so probably does not matter how we subsample here?

Reply:
Probably not, unless we hit issues with methods that expect a minimum number of cells from each celltype.

Reply:
As this is a data loader I could also imagine that a batch-aware task requires multiple batches or a minimum number of cells to be present per batch. I would probably worry less about cell type... and then make a note in the code about this so it could be amended if this becomes an issue in future tasks.

Reply:
I lean towards just using `openproblems.data.utils.subsample_even(adata, n_obs=500, even_obs="method")`